torch/_inductor/triton_heuristics.py

```diff
@@ -605,14 +714,32 @@ def reduction(size_hints, reduction_hint=False, meta=None, filename=None):
         if config.max_autotune or config.max_autotune_pointwise:
             pass  # skip all these cases
         elif reduction_hint == ReductionHint.INNER:
-            return cached_autotune([contiguous_config], meta=meta)
+            return cached_autotune(
+                [contiguous_config],
+                meta=meta,
+                heuristic_type=HeuristicType.REDUCTION,
+                filename=filename,
```

Comment:
why did we not need to pass filename previously?

Reply:
So previously if there is a single config passed in, we don't need do any tuning. But now even if there is a single config, we may still want to do the coordinate descent tuning. We want to write the tuning results to the cache.

The filename is passed in since the cache path is computed based on filename.